scilifelab/lims_utils/workset_upload.py

```diff
@@ -63,42 +76,80 @@ def __init__(self, lims, crawler):
         outs=crawler.starting_proc.all_outputs()
         for out in outs:
             if out.type == "Analyte" and len(out.samples) == 1 :
-                self.name.add(out.location[0].name)
-        self.obj['name']=self.name.pop()
+                try:
+                    self.name.add(out.location[0].name)
+                except:
+                    self.log.warn("no name found for workset {}".format(out.id))
+
+        try:      
+            self.obj['name']=self.name.pop()
+        except:
+            self.log.error("No name found for current workset {}, might be an ongoing step.".format(crawler.starting_proc.id))
+            raise NameError
+        self.obj['technician']=crawler.starting_proc.technician.initials
         pjs={}
         for p in crawler.projects:
             pjs[p.id]={}
             pjs[p.id]['name']=p.name
-            if "library" in p.udf:
+            try:
                 pjs[p.id]['library']=p.udf['Library construction method']
```

Comment:
because lims udfs are entitydescriptors described in the genologics.entities package, and not stdlib objects. That means some of the methods you might not be familiar with are not available on them.